File: dailyQuestion/2020-07/07-01/golang/solution_baoli.go
package main

import "fmt"

func findLength(A []int, B []int) int {
	len_A := len(A)
	len_B := len(B)
	result := 0
	for i := 0; i < len_A; i++ {
		for j := 0; j < len_B; j++ {
			k := 0
			for i+k < len_A && j+k < len_B && A[i+k] == B[j+k] {
				k++
			}
			if k > result {
				result = k
			}
		}
	}
	return result
}

func main() {
	A := []int{1, 2, 3, 2, 1}
	B := []int{3, 2, 1, 4, 7}
	result := findLength(A, B)
	fmt.Println(result)
}

File: dailyQuestion/2020/2020-07/07-01/python/solution_baoli.py
def findLength(A: [int], B: [int]) -> int:
    len_A = len(A)
    len_B = len(B)
    min_len = min(len_A, len_B)
    result = 0
    for i in range(len_A) :
        for j in range(len_B) :
            k = 0
            while i + k < min_len and j+k < min_len and A[i+k] == B[j+k] :
                k += 1
            result = max(result, k )
    return result

if __name__ == "__main__" :


    A = [1,2,3,2,1]
    B = [3,2,1,4,7]

    a = A[0:1] == B[2:3]
    result = findLength(A,B)
    print(result)
